spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/actions/TestExpireSnapshotsAction.java

```diff
@@ -1147,7 +1147,7 @@ public void testUseLocalIterator() {
       checkExpirationResults(1L, 0L, 0L, 1L, 2L, results);
 
       Assert.assertEquals("Expected total number of jobs with stream-results should match the expected number",
-          5L, jobsRunDuringStreamResults);
+          4L, jobsRunDuringStreamResults);
```

Comment:
Spark is planning the scan more optimally by reusing BatchScanExecs, so there are only 4 tasks. However if anyone has more in depth knowledge of this assertion, it would be good to double check this is correct.

Reply:
could you explain what the 4 task is? I'm really confused, thanks a lot